tensorflow/contrib/learn/python/learn/tests/test_saver.py

```diff
@@ -83,6 +83,20 @@ def testNoCheckpoints(self):
     with self.assertRaises(ValueError):
       learn.TensorFlowEstimator.restore(path)
 
+  def testOverwriteParams(self):
+    path = tf.test.get_temp_dir() + '/tmp_saver3'
+    random.seed(42)
+    iris = datasets.load_iris()
+    classifier = learn.TensorFlowDNNClassifier(hidden_units=[10, 20, 10],
+                                               n_classes=3,
+                                               learning_rate=0.1)
+    classifier.fit(iris.data, iris.target)
+    classifier.save(path)
+    self.assertEqual(classifier.learning_rate, 0.1)
+    new_classifier = learn.TensorFlowEstimator.restore(path,
+      new_params={'learning_rate': 0.2})
+    self.assertEqual(type(new_classifier), type(classifier))
+    self.assertEqual(new_classifier.learning_rate, 0.2)
```

Comment:
that said - there is a lot of other arguments changing will lead to problems, for example if one changes hidden_units or n_classes - it will lead to a mismatch with checkpoints and etc.

I would instead recommend creating new Estimator with new params and then just restoring model variables from existing checkpoints. see `learn/utils/checkpoints` library for an `init_from_checkpoints` function that can help with loading. We can add a nice interface for it on Estimator (not sure how it should look yet though), but already one can use it to restore parameters from another trained estimator.